src/Microsoft.ComponentDetection.Contracts/TypedComponent/CondaLockComponent.cs

```diff
@@ -0,0 +1,47 @@
+namespace Microsoft.ComponentDetection.Contracts.TypedComponent;
+
+using System.Collections.Generic;
+
+public class CondaLockComponent : TypedComponent
+{
+    public CondaLockComponent(string name, string version, string category, Dictionary<string, string> dependencies, Dictionary<string, string> hash, string manager, bool optional, string platform, string url)
+    {
+        this.Name = this.ValidateRequiredInput(name, nameof(this.Name), nameof(ComponentType.Conda));
+        this.Version = this.ValidateRequiredInput(version, nameof(this.Version), nameof(ComponentType.Conda));
+        this.Category = category;
+        this.Dependencies = dependencies;
```

Comment:
The component contracts should not contain the dependencies themselves, that is responsibility of the `ISingleFileComponentRecorder`, see how we are leveraging `singleFileComponentRecorder.RegisterUsage` method calls, where we specify the current component we found + the parent component.

Reply:
Great call out! 

As discussed in another comment, this has been refactored and the existing Conda contract is used now while passing along the minimal amount of information that are required for component governance (similar to how PipComponent is doing this.).